src/config.rs

```diff
@@ -1330,11 +1330,17 @@ impl TiKvConfig {
         }
 
         if last_cfg.storage.data_dir != self.storage.data_dir {
-            return Err(format!(
-                "storage data dir have been changed, former data dir is {}, \
-                 current data dir is {}, please check if it is expected.",
-                last_cfg.storage.data_dir, self.storage.data_dir
-            ));
+            // In v3 the default value of storage.data-dir changed
+            // from "" to "./"
+            let using_default_after_upgrade =
+                last_cfg.storage.data_dir == "" && self.storage.data_dir == "./";
```

Comment:
Could we change it like  `if data_dir.is_empty() or data_dir == DEFAULT_DATA_DIR`? I think it's more clear for the reviewers

Reply:
Yes, nice suggestion.